File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/560-SubarraySumEqualsK/560-SubarraySumEqualsK.cpp
// Last updated: 8/4/2026, 7:15:07 PM
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int cnt = 0 ;
        int n = nums.size();
        for(int i = 0 ; i < n ; i++){
            int sum = 0;
            for(int j = i ; j < n ; j++){
                sum += nums[j];

                if(sum == k) cnt++;
            }
        }
        return cnt;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/560-SubarraySumEqualsK/560-SubarraySumEqualsK.java
// Last updated: 8/4/2026, 7:15:00 PM
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n = nums.length;
        int count = 0;
        Map<Integer , Integer>mpp = new HashMap<>();
        mpp.put(0 , 1);
        int l = 0;
        int r = 0;
        int sum = 0;
        while(r < n){
            sum += nums[r];

            if(mpp.containsKey(sum - k)){
                count += mpp.get(sum - k);
            }
            mpp.put(sum, mpp.getOrDefault(sum, 0) + 1);
            r++;
        }
        return count;
    }
}